common/plugins/org.jboss.tools.common.ui/src/org/jboss/tools/common/ui/JobResultFuture.java

```diff
@@ -41,12 +41,13 @@ public JobResultFuture(Job job) {
 
 	@Override
 	public boolean cancel(boolean mayInterruptIfRunning) {
-		if ((isRunning(job)
-				&& mayInterruptIfRunning)
-				|| !isRunning(job)) {
```

Comment:
State 'sleeping' may mean UI jobs deadlock, the ability to cancel a sleeping job is the true way to salvation.